File: src/main/php/Easy/RomanToInt.php
<?php

declare(strict_types=1);

namespace Src\Easy;

class RomanToInt
{
    public static $roman_zero = ['N', 'nil'];

    public function romanToInt(string $roman): int
    {
        $romanNumerals = [
            'I' => 1,
            'V' => 5,
            'X' => 10,
            'L' => 50,
            'C' => 100,
            'D' => 500,
            'M' => 1000,
        ];

        $result = 0;
        $prevValue = 0;
        if (strlen($roman) < 1) {
            return 0;
        }
        for ($i = strlen($roman) - 1; $i >= 0; $i--) {
            $currentValue = array_key_exists($roman[$i], $romanNumerals)
                ? $romanNumerals[$roman[$i]]
                : 0;
            if ($currentValue < $prevValue) {
                $result -= $currentValue;
            } else {
                $result += $currentValue;
            }

            $prevValue = $currentValue;
        }

        return $result;
    }
}

File: src/main/java/easy/RomanToInt.java
package easy;

import java.util.Map;

public class RomanToInt {

  private static final Map<Character, Integer> romanNumerals =
      Map.of('I', 1, 'V', 5, 'X', 10, 'L', 50, 'C', 100, 'D', 500, 'M', 1000);

  public int toInt(String roman) {
    int total = 0;
    int end = roman.length();

    for (int i = 0; i < end; i++) {
      int currentValue = romanNumerals.get(roman.charAt(i));
      int nextValue = (i < end - 1) ? romanNumerals.get(roman.charAt(i + 1)) : 0;
      if (currentValue < nextValue) {
        total -= currentValue;
      } else {
        total += currentValue;
      }
    }
    return total;
  }

  public int solve(String roman) {
    return this.toInt(roman);
  }
}
